tests/testthat/test-assertions.R

```diff
@@ -70,3 +81,17 @@ test_that("validate_rows throws a message if there are no columns selected", {
 
   expect_message(validate_rows(validation, rowSums, function(x) TRUE))
 })
+
+test_that("validation returns assert_success or assert_errors attribute based on result", {
+  data <- data.frame(
+    V1 = c(1, 0),
+    V2 = c(-1, -2)
+  )
+  data <- validate(data)
+
+  val_success <- validate_if(data, V2 < 0)
+  val_error <- validate_if(data, V1 < 0)
+
+  expect_true("assertr_success" %in% names(attributes(val_success)))
+  expect_true("assertr_errors" %in% names(attributes(val_error)))
```

Comment:
We might check as well here, that `assertr_errors` does not appear when there is a success (and the other way round). That will clarify the situation in those cases. Some people might expect that `assertr_errors` is always there, but e.g. NULL in case of success.

Reply:
thank you for your suggestion @mdubel. I've added the expectations in [b65852b](https://github.com/Appsilon/data.validator/pull/59/commits/b65852b693d3de366c4e8e08d928d36b16497f97).